### DISCUSSION TITLE
Developer: yarn run start -> Kubernetes API token

### DISCUSSION THREAD

**Discussion by SohrabTa** [kubeapps/kubeapps#4953] [Q&A] [answered]
I want to extend some functionality of Kubeapps. For that I completed [this](https://github.com/vmware-tanzu/kubeapps/blob/main/site/content/docs/latest/reference/developer/dashboard.md) tutorial to setup the development environment. During that process I clean installed Kubeapps on a new minikube cluster like [this](https://github.com/vmware-tanzu/kubeapps/blob/main/site/content/docs/latest/tutorials/getting-started.md).
 
When I run `yarn run start` after setting up telepresence I get the following success message:
```
Compiled successfully!

You can now view dashboard in the browser.

  Local:            http://localhost:3000
  On Your Network:  http://10.208.81.77:3000

Note that the development build is not optimized.
To create a production build, use yarn build.

webpack compiled successfully
Files successfully emitted, waiting for typecheck results...
Issues checking in progress...
No issues found.
```

When I open the dashboard by opening `http://localhost:3000` in my firefox browser I get prompted to enter the Kubernetes API token as usual. When I run `[Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($(kubectl get --namespace default secret kubeapps-operator-token -o jsonpath='{.data.token}')))` inside the powershell it returns a token, but when I paste this token into the Kubeapps login page it is not accepted. It says `There was an error connecting to the Kubernetes API. Please check that your token is valid.` I made sure that I paste correctly and the token is pasted in its entirety. 

Do I have to use another token when redirecting traffic with telepresence or is there a default token that I have to use to enter the Kubeapps dashboard.

**Comment by castelblanque** [comment, marked as answer]
Hi @SohrabTa , thanks for working with Kubeapps.
If you think what you are going to develop is a feature that the community could enjoy, please don't hesitate to share it with a PR!

The token you use should be a valid one directly against the Kubernetes API server. You can test it with e.g.:

```bash
curl $APISERVER/openapi/v2  --header "Authorization: Bearer $TOKEN" --cacert ca-cert.crt
```

You could also try skipping Telepresence and use the React proxy as you might have [seen here](https://github.com/vmware-tanzu/kubeapps/blob/main/site/content/docs/latest/reference/developer/dashboard.md#telepresence-alternative).

**Reply by SohrabTa** [reply]
I was not able to make telepresence work, but the React proxy way worked well after some tinkering! Thanks a lot.

**Reply by castelblanque** [reply]
Glad to hear that it worked!
